bot/exts/evergreen/trivia_quiz.py

```diff
@@ -238,40 +240,56 @@ def cog_unload(self) -> None:
+        """Get yesterday's most read articles from wikipedia and format them like trivia questions."""
         error_fetches = 0
         wiki_questions = []
         # trivia_quiz.json follows a pattern, every new category starts with the next century.
         start_id = 501
+        yesterday = datetime.strftime(datetime.now() - timedelta(1), '%Y/%m/%d')
 
-        while len(wiki_questions) != 10 and error_fetches < MAX_ERROR_FETCH_TRIES:
-            async with self.bot.http_session.get(url=RANDOM_WIKI_URL) as r:
+        while error_fetches < MAX_ERROR_FETCH_TRIES:
+            async with self.bot.http_session.get(url=WIKI_FEED_API_URL.format(date=yesterday)) as r:
                 if r.status != 200:
                     error_fetches += 1
                     continue
-
-                raw_data = await r.json()
-
-                if not all(c.isalnum() for c in raw_data["title"]):
-                    # If the title contains non alphabet/numerical(s) then
-                    # don't append it to the question and move on to the next.
-                    # Eg. "Knight's night tour" would be skipped.
-                    continue
-
-                formatted_data = {
-                    "id": start_id,
-                    "question": f"Guess the title of the Wikipedia article.\n\n"
-                                # Sometimes the wikipedia title is given in the extract,
-                                # giving away the answers, to remove then, they are replaced
-                                # with '[redacted]'.
-                                f"{raw_data['extract'].replace(raw_data['title'], '[redacted]')}",
-                    "answer": raw_data["title"],
-                    "info": raw_data["extract"]
-                }
-                start_id += 1
-                wiki_questions.append(formatted_data)
+                raw_json = await r.json()
+                articles_raw = raw_json["mostread"]["articles"]
+
+                for article in articles_raw:
+                    # Form the question, strategy:
+                    # - Check if `extract` is present in the article, if it doesn't then skip the article
+                    # Sometimes, the wikipedia title is given in the extract, therefore giving away the answer
+                    # therefore, we need to remove all cases of it
+                    question = article.get("extract")
+                    # multi letter words are formatted this way: Jason_Terror
+                    title = article["normalizedtitle"].replace("_", " ")
+
+                    if not question:
+                        continue
+                    elif not all(c.isalnum() for c in title):
+                        # If the title contains non alphabet/numerical(s) then
+                        # don't append it to the question and move on to the next.
+                        # Eg. "Jurassic World: Dominion (2022)" would be skipped.
+                        continue
```

Comment:
Umm, yeah I will change this to special characters like `~!@#$%^&*(){}[]<>?/:";'` otherwise you get answers as `Jurassic Park (Dominion)`, which would be marked incorrect when you answer it without them.

Reply:
shouldn't the fuzzy matching of answers take care of that?

Reply:
For some cases it does, for some, it doesn't.

Reply:
I think tweaking the fuzzy match tolerance to make it a bit more forgiving may be a better solution than this, I reckon false matches would be fairly rare so we could probably be relatively forgiving. Thoughts?

Reply:
I'm not sure this was resolved. If we think it's hard to guess because of the punctuation we could give the punctuation with the words redacted before the description, this would also help people know the order/length of words, could maybe do that the same as how we do it from the extract
```python
cleaned_title = title
for word in re.split("[\s-]", title):
    word = word.strip(string.punctuation)
    secret_word = "\*"*len(word)
    cleaned_title = re.sub(rf'\b{word}\b', f"**{secret_word}**", cleaned_title, flags=re.IGNORECASE) 
```
Haven't tested this or thought if there might be a cleaner way though